### PR TITLE
Add Healthcheck update test

### PR DESCRIPTION
**Pull request by akwi-github**
Setup
 1) Create tcproute
 2) Create backend service
 3) Create healthcheck for backend service
 4) Connect backend service with tcproute

Changes
 1) Update healthcheck

Expectations
 1) Verify that action succeeded

b/332473784

### CODE REVIEW

File: e2e/healthcheck_update_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+					Address: "10.240.15.83/32",
+					Port:    "80",
+				},
+			},
+		},
+	}
+
+	tcprID, err := buildTCPRoute(graphBuilder, "hc-update-test", meshURL, rules, bsID)
+	if err != nil {
+		t.Fatalf("buildTCPRoute(_, tcproute-test, _, _, _) = (_, %v), want (_, nil)", err)
```
Comment (kl52752) [inline e2e/healthcheck_update_test.go:71]: name mismatch: should be hc-update-test

File: e2e/healthcheck_update_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	expectedActions := []exec.ActionMetadata{
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, tcprID)},
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, bsID)},
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, hcID)},
+		{Type: exec.ActionTypeCreate, Name: actionName(exec.ActionTypeCreate, negID)},
+	}
+	processGraphAndExpectActions(t, graphBuilder, expectedActions)
+
+	checkGCEHealthCheck(t, ctx, theCloud, hcID, 15)
+	hcID, err = buildHealthCheck(graphBuilder, "hc-update-test", 25)
```
Comment (kl52752) [inline e2e/healthcheck_update_test.go:88]: please add comment
// Update health check

File: e2e/healthcheck_update_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+	expectedActions = []exec.ActionMetadata{
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, hcID)},
+		{Type: exec.ActionTypeUpdate, Name: actionName(exec.ActionTypeUpdate, tcprID)},
```
Comment (kl52752) [inline e2e/healthcheck_update_test.go:92]: this should not happened :)

File: e2e/healthcheck_update_test.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+	t.Cleanup(func() {
+		err := theCloud.NetworkEndpointGroups().Delete(ctx, negID.Key)
+		t.Logf("theCloud.NetworkEndpointGroups().Delete(ctx, %s): %v", negID.Key, err)
+	})
+
+	hcID, err := buildHealthCheck(graphBuilder, "hc-update-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc-update-test, 15) = (_, %v), want (_, nil)", err)
+	}
+	t.Cleanup(func() {
```
Comment (kl52752) [inline e2e/healthcheck_update_test.go:37]: health check, backend service and tcproutes (all resources that are managed by rGraph) cleanup can be in one function after executor run

File: e2e/healthcheck_update_test.go (hunk 3 above)
Comment (AwesomePatrol) [inline e2e/healthcheck_update_test.go:92]: What's the diff that causes an update here?

**Review by AwesomePatrol — APPROVED** [review]

**Comment by bowei** [comment]
Hmm -- I didn't want this to merge, but allow others to LGTM to merge this.

Please make the changes specified in the comments. 

The "approve" was not meant to skip code review.

I will make sure to use the "hold" from now on.

File: e2e/healthcheck_update_test.go (hunk 1 above)
Comment (akwi-github) [reply e2e/healthcheck_update_test.go:71]: Done.

File: e2e/healthcheck_update_test.go (hunk 2 above)
Comment (akwi-github) [reply e2e/healthcheck_update_test.go:88]: Done.

File: e2e/healthcheck_update_test.go (hunk 4 above)
Comment (akwi-github) [reply e2e/healthcheck_update_test.go:37]: Done.

File: e2e/healthcheck_update_test.go (hunk 3 above)
Comment (akwi-github) [reply e2e/healthcheck_update_test.go:92]: Added TODO.

Comment (akwi-github) [reply e2e/healthcheck_update_test.go:92]: Added TODO.

**Pull request merged**